packages/nestjs-module/src/ogma.service.ts

```diff
@@ -50,64 +64,77 @@ export class OgmaService implements LoggerService {
-    this.printMessage(message, 'debug', context);
+  public debug(message: any, context?: string, requestId?: string): void {
+    this.printMessage(message, 'debug', context, requestId);
   }
 
   /**
    * use Ogma to log at the FATAL level
    * @param message What to print to the Ogma instance
    * @param context Optional context if you want to change what the original context was
+   * @param requestId Optional id of an request
    */
-  public fatal(message: any, context?: string): void {
-    this.printMessage(message, 'fatal', context);
+  public fatal(message: any, context?: string, requestId?: string): void {
+    this.printMessage(message, 'fatal', context, requestId);
   }
 
   /**
    * use Ogma to log at the SILLY level
    * @param message What to print to the Ogma instance
    * @param context Optional context if you want to change what the original context was
+   * @param requestId Optional id of an request
    */
-  public silly(message: any, context?: string): void {
-    this.printMessage(message, 'silly', context);
+  public silly(message: any, context?: string, requestId?: string): void {
+    this.printMessage(message, 'silly', context, requestId);
   }
 
   /**
    * use Ogma to log at the VERBOSE level
    * @param message What to print to the Ogma instance
    * @param context Optional context if you want to change what the original context was
+   * @param requestId Optional id of an request
    */
-  public verbose(message: any, context?: string): void {
-    this.printMessage(message, 'verbose', context);
+  public verbose(message: any, context?: string, requestId?: string): void {
+    this.printMessage(message, 'verbose', context, requestId);
   }
 
   /**
    * A predefined method for printing errors to the Ogma instance
    * @param error The error to print. Should be an Error or Exception object
    * @param context Optional context if you want to change what the original context was
+   * @param requestId Optional id of an request
    */
-  public printError(error: Error, context?: string): void {
-    this.printMessage('', 'error', context);
+  public printError(error: Error, context?: string, requestId?: string): void {
+    this.printMessage('', 'error', context, requestId);
+    if (this.requestContext && this.context) {
+      requestId = this.getRequestId(this.requestContext);
+    }
     context = context ?? this.context;
-    this.ogma.printError(error, context);
+    this.ogma.printError(error, context, undefined, requestId);
   }
 
   private printMessage(
     message: any,
     levelString: Exclude<keyof Ogma, 'printMessage' | 'printError'>,
     context?: string,
+    requestId?: string,
   ): void {
     context = context ?? this.context;
-    this.ogma[levelString](message, context);
+    if (this.requestContext && this.context) {
+      requestId = this.getRequestId(this.requestContext);
+    }
+    this.ogma[levelString](message, context, undefined, requestId);
```

Comment:
What happens if a requestId is passed but the `OgmaService` also has a `this.context` and `this.requestContext`? Would that ever be a valid use case?

Reply:
Yeah, I will add there third condition with checking if `requestId` is not passed. Right now it will replace it in that case